docs/code-review-guide.md

```diff
@@ -0,0 +1,33 @@
+# Yeti Code Review Guide
+
+This document is intended to be a reference for Yeti developers from the perspective of both the code reviewer and the person submitting code. We understand that there may be strong personal preferences on this topic, so please treat this document as a set of guidelines to follow rather than a strict set of rules.
+
+## Purpose of Code Review
+
+We always aim to deliver a high-quality product to our clients. Code reviews are a critical part of this process. Having our work reviewed by peers and superiors ensures that we maintain accountability amongst the team and this ultimately drives client satisfaction.
+
+## Priorities for Code Reviewers
+
+While code review mainly deals with product quality, it's also important that we deliver the product in a timely fashion. For this reason, reviewers should exercise their judgement on how to best prioritize the various considerations during review.
+
+Regarding whether to request changes or to approve with feedback, we suggest that you request changes if items 1-2 are not satisfied. Otherwise, it is up to the reviewer. Keep in mind that if you approve with feedback, you are essentially leaving it up to the submitter whether and when to incorporate your suggestions.
```

Comment:
There's a third option here, which is to leave a comment instead of approving or requesting changes.

I will sometimes do this when I have questions or comments about a piece of code, and I'd like to hear from the author before deciding whether to request changes or approve.

Reply:
Great point - I'll add a sentence or two on this